yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,125 @@
+description: >
+    Implement to manage VPD data in system.
+methods:
+    - name: WriteKeyword
+      description: >
+          A method to update the value of a keyword in a given VPD file. Keyword
+          can be updated on hardware or it can be updated on both hardware and
+          cache. To update on hardware the target should be 1 and to update on
+          both hardware and cache the target is 2.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Path of the FRU whose keyword need to be upadted.
+          - name: keyword
+            type: struct[string,string,array[byte]]
+            description: >
+                It is a tuple of record, keyword and value that is to be
+                updated.
+          - name: target
+            type: size
+            description: >
+                Target where the keyword needs to be updated.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - com.ibm.VPD.Error.PathNotFound
+          - com.ibm.VPD.Error.RecordNotFound
+
+    - name: ReadKeyword
+      description: >
+          An api to read VPD keyword. Keyword can be read from hardware or from
+          cache. To read keyword from cache the target should be 0 and to read
+          from hardware the target is 1.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Path of the FRU whose keyword needs to be read.
+          - name: Keyword
+            type: struct[string,string]
+            description: >
+                It is a tuple of record and keyword to be read.
+          - name: target
+            type: size
+            description: >
+                Target where the keyword to be read from cache/hardware.
+      returns:
+          - name: value
+            type: array[byte]
+            description: >
+                value of the corresponding keyword.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - com.ibm.VPD.Error.PathNotFound
+          - com.ibm.VPD.Error.RecordNotFound
+
+    - name: GetHardwarePath
+      description: >
+          An api to get hardware path for the corresponding FRU DBus object
+          path.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                DBus object path of the FRU whose hardware path needs to be
+                collected.
+      returns:
+          - name: Path
+            type: string
+            description: >
+                Hardware path of the corresponding FRU.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
```

Comment:
can you remove Invalid argument and add com.ibm.VPD.Error.PathNotFound

Reply:
invalid argument is valid as well as other error types needs to be set if any. think on this more